src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -180,34 +184,80 @@ public void validateUpdate(final Integer locationId, final LocationRequestDto lo
-		final Integer countryId = locationRequestDto.getCountryId() != null ? locationRequestDto.getCountryId() : locationDTO.getCountryId();
-		final Integer provinceId = locationRequestDto.getProvinceId() != null ? locationRequestDto.getProvinceId() : locationDTO.getProvinceId();
+
+		final Integer countryId = locationRequestDto.getCountryId();
+		final Integer provinceId = locationRequestDto.getProvinceId();
 		this.validateCountryAndProvince(countryId, provinceId);
 	}
 
 	public void validateCanBeDeleted(final Integer locationId) {
 		this.errors = new MapBindingResult(new HashMap<>(), Integer.class.getName());
 
 		this.validateLocation(this.errors, locationId);
+		this.validateLocationNotDeletable(locationId);
 		this.validateLocationNotUsedInGermplasm(locationId);
 		this.validateLocationNotUsedInLot(locationId);
 		this.validateLocationNotUsedInAttribute(locationId);
 		this.validateLocationNotUsedInName(locationId);
 		this.validateLocationNotUsedInStudy(locationId);
+		this.validateLocationNotBelongToCountryTable(locationId);
+		this.validateLocationNotUsedInFieldMap(locationId);
+
+	}
+
+	private void validateLocationNotDeletable(final Integer locationId) {
+		final LocationDTO location = this.locationService.getLocation(locationId);
+
+		if (LOCATIONS_NOT_DELETABLES.contains(location.getName()) || location.isDefaultLocation()) {
+			this.errors.reject("location.not.deletable", new String[] {locationId.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationNotEditable(final Integer locationId) {
+		final LocationDTO location = this.locationService.getLocation(locationId);
+
+		if (LOCATIONS_NOT_DELETABLES.contains(location.getName())) {
+			this.errors.reject("location.not.editable", new String[] {locationId.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationNotUsedInFieldMap(final Integer locationId) {
+		final Location location = locationDataManager.getLocationByID(locationId);
+		List<Locdes> blocks = null;
+		if(location.getLtype() == 415){ // Field
+			blocks = locationDataManager.getLocdes(null, locationId.toString());
+		}
+		if(location.getLtype() == 416){ // Block
+			blocks = locationDataManager.getLocdes(locationId, null).stream().filter(locdes -> locdes.getTypeId() == 313)
+				.collect(Collectors.toList());
+
+		}
+		if (!CollectionUtils.isEmpty(blocks)) {
+			final List<Integer> locationIds = blocks.stream().map((locdes) -> locdes.getLocationId()).collect(Collectors.toList());
+			boolean isUsed = locationService.blockIdIsUsedInFieldMap(locationIds);
+			if (isUsed) {
+				this.errors.reject("location.is.used.in.field.map", new String[] {locationId.toString()}, "");
+				throw new ApiRequestValidationException(this.errors.getAllErrors());
+			}
+		}
+
+	}
+
+	private void validateLocationNotBelongToCountryTable(final Integer locationId) {
```

Comment:
@cuenyad rename to validateLocationIsNotDefaultCountry

Reply:
done.